### PR TITLE
Run3-alca218D Clean some of the remaining Hcal calibration codes in Calibration/HcalCalibAlgos

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Clean some of the remaining Hcal calibration codes in Calibration/HcalCalibAlgos

#### PR validation:

Use some of the test workflows in Calibration/HcalCalibAlgos/test/python

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

File: Calibration/HcalCalibAlgos/plugins/RecAnalyzerMinbias.cc [hunk 1]
```diff
@@ -504,8 +499,7 @@ void RecAnalyzerMinbias::analyze(const edm::Event& iEvent, const edm::EventSetup
     h_AmplitudeHF_->Fill(amplitudefullHF);
   }
 
-  edm::Handle<HBHERecHitCollection> hbheMB;
-  iEvent.getByToken(tok_hbherecoMB_, hbheMB);
+  const edm::Handle<HBHERecHitCollection>& hbheMB = iEvent.getHandle(tok_hbherecoMB_);
   if (!hbheMB.isValid()) {
     edm::LogWarning("RecAnalyzer") << "HcalCalibAlgos: Error! can't get hbhe product!";
```
Comment (tvami) [inline Calibration/HcalCalibAlgos/plugins/RecAnalyzerMinbias.cc:504]:
```suggestion
    edm::LogWarning("RecAnalyzerMinbias") << "HcalCalibAlgos: Error! can't get hbhe product!";
```
This is the correct module name

File: Calibration/HcalCalibAlgos/plugins/SimAnalyzerMinbias.cc [hunk 2]
```diff
@@ -147,8 +143,7 @@ void SimAnalyzerMinbias::analyze(const edm::Event& iEvent, const edm::EventSetup
                                    << myGenEvent->vertices_size() << " vertices";
   }
 
-  edm::Handle<edm::PCaloHitContainer> hcalHits;
-  iEvent.getByToken(tok_hcal_, hcalHits);
+  const edm::Handle<edm::PCaloHitContainer>& hcalHits = iEvent.getHandle(tok_hcal_);
   if (!hcalHits.isValid()) {
     edm::LogWarning("AnalyzerMB") << "Error! can't get HcalHits product!";
```
Comment (tvami) [inline Calibration/HcalCalibAlgos/plugins/SimAnalyzerMinbias.cc:148]:
```suggestion
    edm::LogWarning("SimAnalyzerMinbias") << "Error! can't get HcalHits product!";
```

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
Hi @bsunanda in `Calibration/HcalCalibAlgos/plugins/HOCalibAnalyzer.cc` there is a dead assignment of `tmpsect`, please remove it.

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by tvami** [comment]
+alca

**Comment by perrotta** [comment]
+1

**Pull request merged**